aws/resource_aws_s3_bucket_test.go

```diff
@@ -126,6 +126,27 @@ func TestAccAWSS3Bucket_region(t *testing.T) {
 	})
 }
 
+func TestAccAWSS3Bucket_bucketRegionalDomainName(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSS3BucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSS3BucketConfigRegionalDomainName(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3BucketExists("aws_s3_bucket.bucket"),
+					resource.TestCheckResourceAttr("aws_s3_bucket.bucket", "region", "ap-south-1"),
+					resource.TestCheckResourceAttr(
```

Comment:
As mentioned in the other PR with a `Computed: true` attribute that is always available, this can be moved as a single check under the `_basic` test.

We also explicitly need the S3 testing to be able to run from AWS Commercial, AWS GovCloud (US), and AWS China which means that we cannot hard code a specific partition region. I believe there is a `testAccProviderRegion()` helper function which can aid us here.

If we want to ensure that specific regions are returning their expected domain names (e.g. `cn-north-1` actually returning `BUCKET.s3.cn-north-1.amazonaws.com.cn`) I would highly recommend setting up unit testing for `bucketRegionalDomainName` as well.

Reply:
Since I need to create separate test config, I used separate acceptance test `TestAccAWSS3Bucket_bucketRegionalDomainName`. I couldn't find `testAccProviderRegion()`, so I used the same `endpoints` package to choose region.